File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/CoachResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.*;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CoachResponseDTO {
    private Long id;
    private String firstName;
    private String lastName;
    private Integer age;
    private LocationDTO location;
    private GenderType gender;
    private String phoneNumber;
    private String email;
    private CoachSpecialization specialization;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Double rate;
    private VisibilityStatus profileVisibility;
    private MessagePermission messagePermission;
    private AccountPrivacy accountPrivacy;
    private String expertise;
    private String services;
    private String availability;
    private LocalDateTime lastSeen;
}

File: OrienGO-Frontend/src/app/model/dto/CoachResponse.dto.ts
import { AccountPrivacy } from "../enum/AccountPrivacy.enum";
import { CoachSpecialization } from "../enum/CoachSpecialization.enum";
import { GenderType } from "../enum/GenderType.enum";
import { MessagePermission } from "../enum/MessagePermission.enum";
import { VisibilityStatus } from "../enum/VisibilityStatus.enum";
import { LocationDTO } from "./Location.dto";

export interface CoachResponseDTO {
  id: number;
  firstName: string;
  lastName: string;
  age: number;
  gender: GenderType;
  phoneNumber?: string;
  email: string;
  specialization?: CoachSpecialization

  profileVisibility: VisibilityStatus;
  messagePermission: MessagePermission;
  accountPrivacy: AccountPrivacy;

  expertise?: string; // e.g., "Career Development, Personal Growth, Goal Setting"
  services?: string; // e.g., "Available for 1-on-1 sessions"
  availability?: string; // e.g., "Monday - Friday: 9 AM - 6 PM; Weekend: By appointment"

  createdAt?: string; // ISO string from backend
  updatedAt?: string; // ISO string from backend

  location: LocationDTO;
  lastSeen?: string;
  rate?: number;
}